src/storage/IndexBaseProcessor.inl

```diff
@@ -0,0 +1,188 @@
+        INDEX_CREATE_DONE;
+    }
+
+    while (iter->valid()) {
+        key = iter->key().str();
+        val = iter->val().str();
+        iter->next();
+        indexKey = (indexType_ == nebula::cpp2::IndexType::EDGE) ?
+                   assembleEdgeIndexKey(spaceId_, partId, key, std::move(val)) :
+                   assembleVertexIndexKey(spaceId_, partId, key, std::move(val));
+
+        if (!indexKey.ok()) {
+            continue;
+        }
+        data.emplace_back(std::move(indexKey.value()), std::move(key));
+        batchPutNum++;
+        if (batchPutNum == FLAGS_bulk_number_per_index_creation) {
+            auto code = doBatchPut(spaceId_, partId, data);
+            if (code == cpp2::ErrorCode::E_LEADER_CHANGED) {
+                nebula::cpp2::HostAddr leader;
+                auto addrRet = this->kvstore_->partLeader(spaceId_, partId);
+                CHECK(ok(addrRet));
+                auto addr = value(std::move(addrRet));
+                leader.set_ip(addr.first);
+                leader.set_port(addr.second);
+                thriftResult.set_leader(leader);
+            } else if (code != cpp2::ErrorCode::SUCCEEDED) {
+                thriftResult.set_code(code);
+                INDEX_CREATE_DONE;
+            }
+            data.clear();
+            batchPutNum = 0;
+        }
+    }
+
+    if (!data.empty()) {
+        auto code = doBatchPut(spaceId_, partId, data);
+        if (code == cpp2::ErrorCode::E_LEADER_CHANGED) {
+            nebula::cpp2::HostAddr leader;
+            auto addrRet = this->kvstore_->partLeader(spaceId_, partId);
+            CHECK(ok(addrRet));
+            auto addr = value(std::move(addrRet));
+            leader.set_ip(addr.first);
+            leader.set_port(addr.second);
+            thriftResult.set_leader(leader);
+        } else if (code != cpp2::ErrorCode::SUCCEEDED) {
+            thriftResult.set_code(code);
+            INDEX_CREATE_DONE;
+        }
+        data.clear();
+    }
+
+    // TODO : send index status 'CONSTRUCTING' via meta client,
+    //        should block all actions of insert|delete| update.
+    //        until index create done of each parts.
+
+    // TODO : Assemble new data changes during index creation.
+
+    INDEX_CREATE_DONE;
+}
+
+template<typename RESP>
+cpp2::ErrorCode IndexBaseProcessor<RESP>::doBatchPut(GraphSpaceID spaceId, PartitionID partId,
+                                          std::vector<kvstore::KV> data) {
+    cpp2::ErrorCode ret = cpp2::ErrorCode::SUCCEEDED;
+    folly::Baton<true, std::atomic> baton;
+    this->kvstore_->asyncMultiPut(spaceId, partId, std::move(data),
+                            [&] (kvstore::ResultCode code) {
+                                if (code != kvstore::ResultCode::SUCCEEDED) {
+                                    ret = this->to(code);
+                                }
+                                baton.post();
+                            });
+    baton.wait();
```

Comment:
is there any method to no call wait(), like future or something else? because wait() will waste a thread.